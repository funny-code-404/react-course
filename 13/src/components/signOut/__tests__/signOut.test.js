import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

import { SignOut } from "..";

configure({ adapter: new Adapter() });

describe('PAGES  SIGN OUT', () => {
    const mockFn = jest.fn();

    const props = {
        singOut: mockFn,
        auth: { user: { name: 'text' }}
    }

    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<SignOut {...props} />);
    });

    it('to Match Snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
    it('component is defined', () => {
        expect(wrapper).toHaveLength(1)
    })
     it('function should be called', () => {
         wrapper.find('button').simulate('click')
        expect(mockFn).toHaveBeenCalledTimes(1)
    })
});
// npm run test src/components/signOut