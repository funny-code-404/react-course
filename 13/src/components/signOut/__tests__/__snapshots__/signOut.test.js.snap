// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PAGES  SIGN OUT to Match Snapshot 1`] = `
<button
  className="sign-out"
  onClick={[Function]}
>
  Sign out
</button>
`;
